symplyphysics/definitions/current_is_charge_derivative.py

```diff
@@ -27,3 +27,12 @@ def calculate_current(charge_start_: Quantity, charge_end_: Quantity, time_: Qua
     dsolved = applied_definition.doit()
     result_expr = dsolved.rhs
     return expr_to_quantity(result_expr, 'current')
+
+@validate_input(current_start_=units.current, current_end_=units.current, time_=units.time)
+@validate_output(units.charge)
+def calculate_charge(current_start_: Quantity, current_end_: Quantity, time_: Quantity) -> Quantity:
+    current_function_ = 0.5 * (current_end_ + current_start_) * time_ * time
```

Comment:
I think the linear function is same as above, so stick to the existing approach.